EC_finetune/__main__.py

```diff
@@ -293,9 +293,38 @@ def main():
     args_dict = vars(args)
     with open(args_dict['config'], 'r') as config_file:
         config_dict = yaml.load(config_file, Loader=yaml.FullLoader)
+
+    # TODO: Fix config parsing from two different sources. With a combination of
+    # reading command line arguments straight into variables they are meant to
+    # override, and applying unconditional dictionary updates, the config's
+    # final values are not working as intended.
+    #
+    # Config file:    { output_dir = foo }
+    # Command line:   {}
+    # Parser default: output_dir_override will update output_dir (default: '')
+    #
+    # Based on the current order of operations,
+    # 1. args { output_dir = '' } & config { output_dir = foo }
+    # 2. args { output_dir = '' } & config { output_dir = '' }
+    # 
+    # We cannot simply swap the following two lines because then overriding
+    # config values via command line arguments does not work as expected.
+    #
+    # In the short term this could be remedied by conditionally applying
+    # dictionary updates if the values have initialized values. In the long term
+    # we plan on moving toward hierarchical config structures which will provide
+    # a single source of truth config to apply to a particular run. At this
+    # point there won't be two competing sources of information to be merged.
+    # Today, we opt to pursue the latter route instead of fixing these config
+    # update bugs which occur throughout the pipeline.
+    # 
+    # As is, this code does not run. Please comment out the next line to allow
+    # the pipeline to run. Note that this removes all command line overrides.
     config_dict.update(args_dict)
     args_dict.update(config_dict)
 
+    set_seed(args.seed, args.n_gpu)
```

Comment:
This fixes the reproducibility issues from the last PR. `set_seed` was dropped when the `add_adapters` was merged [here](https://github.com/CLMBRs/communication-translation/commit/ee81f7f22a6d1b5b6db174efde1dd7007812a76e#diff-58f85cba2e8b9e12750d38907a3ecd5af8560199c22690f77dde94ec1732d34f).

I would expect that T2I would still not reproduce 100% (since it never reproduced on the main branch) but this should get us back to the original I2I results based on my first-pass reproducibility results (assuming `add_adapters` did not introduce subtle side effects).

I'm leaning on not blocking this merge to validate these results (I2I/T2I both take 24+ hours to run). What do you think?

Reply:
Also what README are you referring to? This PR description [link](https://github.com/CLMBRs/communication-translation/pull/13#issue-1380655635)?

Reply:
Sorry, yes, meant the PR description.  Agreed re not blocking!

Reply:
So it turns out that I ran the T2I numbers after the seed was added back so our numbers do not change after a sanity run, and therefore are not reproducing here. This is kinda expected since Agatha was not able to have T2I reproduce when it was first written either.

Reply:
Great! My question was whether our new numbers for T2I are reproducing, not whether we can get the preprint's.  It sounds like they are, which is great!  In other words, we need reproducibility looking forward, not looking backward.